tests/benchmarks/test_tpch.py

```diff
@@ -0,0 +1,315 @@
+import os
+from datetime import datetime
+
+import dask.dataframe as dd
+import pytest
+
+DATASETS = {
+    "scale 100": "s3://coiled-runtime-ci/tpch_scale_100/",
+}
+
+enabled_dataset = os.getenv("TPCH_SCALE")
+if enabled_dataset is not None:
+    if enabled_dataset not in DATASETS:
+        raise ValueError("Unknown tpch dataset: ", enabled_dataset)
+else:
+    enabled_dataset = "scale 100"
+
+
+def read_data(filename):
+    path = DATASETS[enabled_dataset] + filename + "/"
+    return dd.read_parquet(path, engine="pyarrow")
+
+
+@pytest.mark.client("tpch")
+def test_query_1(client):
+    VAR1 = datetime(1998, 9, 2)
+    lineitem_ds = read_data("lineitem")
+
+    lineitem_filtered = lineitem_ds[lineitem_ds.l_shipdate <= VAR1]
+    lineitem_filtered["sum_qty"] = lineitem_filtered.l_quantity
+    lineitem_filtered["sum_base_price"] = lineitem_filtered.l_extendedprice
+    lineitem_filtered["avg_qty"] = lineitem_filtered.l_quantity
+    lineitem_filtered["avg_price"] = lineitem_filtered.l_extendedprice
+    lineitem_filtered["sum_disc_price"] = lineitem_filtered.l_extendedprice * (
+        1 - lineitem_filtered.l_discount
+    )
+    lineitem_filtered["sum_charge"] = (
+        lineitem_filtered.l_extendedprice
+        * (1 - lineitem_filtered.l_discount)
+        * (1 + lineitem_filtered.l_tax)
+    )
+    lineitem_filtered["avg_disc"] = lineitem_filtered.l_discount
+    lineitem_filtered["count_order"] = lineitem_filtered.l_orderkey
+    gb = lineitem_filtered.groupby(["l_returnflag", "l_linestatus"])
+
+    total = gb.agg(
+        {
+            "sum_qty": "sum",
+            "sum_base_price": "sum",
+            "sum_disc_price": "sum",
+            "sum_charge": "sum",
+            "avg_qty": "mean",
+            "avg_price": "mean",
+            "avg_disc": "mean",
+            "count_order": "count",
+        }
+    )
+
+    total.reset_index().compute().sort_values(["l_returnflag", "l_linestatus"])
```

Comment:
nit: feels a little like cheating to compute first before sorting. In a perfect world, dask would know the data is small enough to collect and sort locally

Reply:
That's a good idea, I'll add this to my todo list for dask-expr. I'd like to avoid measuring the intermediate compute though, that's why I added it in this order.

Reply:
sorry, to be more explicit: dask should not sort _locally_ but reduce to a single worker, sort there and return your result